core/tests/protractor_utils/ExplorationEditorSettingsTab.js

```diff
@@ -77,25 +73,27 @@ var ExplorationEditorSettingsTab = function() {
   };
 
   this.enableCorrectnessFeedback = async function() {
-    expect(await enableCorrectnessFeedbackButton.isDisplayed()).toBe(true);
     await action.click(
-      'Enable Correctness Feedback Button', enableCorrectnessFeedbackButton);
+      'Enable Correctness Feedback button', enableCorrectnessFeedbackButton);
   };
 
   this.expectAvailableFirstStatesToBe = async function(names) {
     var options = await initialStateSelect.all(by.tagName('option'))
       .map(async function(elem) {
+        await waitFor.visibilityOf(
+          elem, 'Element ' + elem + ' taking too long to appear');
```

Comment:
Does this actually work? Since `elem` isn't a string, I'm skeptical that this will work correctly. I would suggest replacing lines 81-84 with a `for` loop from `0` to `options.count()`. Then you can use the index in the error message. Does that make sense?